packages/cadl-ranch-specs/http/azure/client-generator-core/flatten/main.tsp

```diff
@@ -0,0 +1,107 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Illustrates the model flatten cases.")
+@scenarioService("/type/model/flatten")
+namespace Type.Model.Flatten;
+
+@doc("This is the model with one level of flattening.")
+model FlattenModel {
+  name: string;
+
+  #suppress "deprecated" "@flattenProperty decorator is not recommended to use."
+  @flattenProperty
+  child: ChildModel;
```

Comment:
It is better if use `properties` and `typeProperties` of real cases.

Reply:
rename `child` to `properties`